tasks/main.yml

```diff
@@ -8,15 +8,19 @@
 
 - name: Gradle | apt-get update
   apt: update_cache=yes
-  
+
 - name: Gradle | Install unzip
   apt: pkg=unzip state=latest
 
 - name: Gradle | Unpack the compressed Gradle
   command: unzip /tmp/gradle-{{gradle_version}}-all.zip -d /usr/local
+  args:
+    creates: /usr/local/gradle-{{gradle_version}}
 
 - name: Grade | Rename Gradle folder
   command: mv /usr/local/gradle-{{gradle_version}} /usr/local/gradle
```

Comment:
Either way is fine as far as I am concerned. The creates statements help
Ansible to identify if the command need to even be attempted.

I think they both achieve the desired goal.

On Mon, Feb 15, 2016 at 1:01 PM Vinay V. Tota <notifications@github.com>
wrote:

> In tasks/main.yml
> <https://github.com/MagneDavidsen/gradle/pull/2#discussion_r52933492>:
>
> >  - name: Gradle | Install unzip
> >    apt: pkg=unzip state=latest
> >
> >  - name: Gradle | Unpack the compressed Gradle
> >    command: unzip /tmp/gradle-{{gradle_version}}-all.zip -d /usr/local
> > +  args:
> > +    creates: /usr/local/gradle-{{gradle_version}}
> >
> >  - name: Grade | Rename Gradle folder
> >    command: mv /usr/local/gradle-{{gradle_version}} /usr/local/gradle
>
> Ok, so I guess there is a problem here, specifically this command is not
> re-runnable
>
> The first time it works as expected, moving
> /usr/local/gradle-{{gradle_version}} to /usr/local/gradle
>
> However once /usr/local/gradle exists it will move
> /usr/local/gradle-{{gradle_version}} into /usr/local/gradle
>
> Change /usr/local/gradle to be a symlink instead?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/MagneDavidsen/gradle/pull/2/files#r52933492>.
>